system_metrics_collector/src/topic_statistics_collector/received_message_period.hpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TOPIC_STATISTICS_COLLECTOR__RECEIVED_MESSAGE_PERIOD_HPP_
+#define TOPIC_STATISTICS_COLLECTOR__RECEIVED_MESSAGE_PERIOD_HPP_
+
+#include <chrono>
+#include <string>
+
+#include "topic_statistics_collector.hpp"
+
+#include "../../src/system_metrics_collector/collector.hpp"
```

Comment:
The CMake should specify `src` as an `include_directory` for this target, rather than doing this. Relative includes are brittle and don't fit the google C++ styleguide - then we can just `#include "system_metrics_collector/collector.hpp"` An alternative would also be to add `src/system_metrics_collector` and include by filename alone